### ISSUE TITLE
Make providing feedback more accessible by making our own forum

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/website-v2#339]
### My suggestion

Having our own feedback forum would create a centralized place for suggestions.

Currently, feedback is spread across a few different places, and on GitHub, it's mixed in with to-dos. With our own feedback forum, community suggestions would be better separated, and everyone would have access to all public feedback and **all in one place**.

Some users don't have a GitHub or Discord account, limiting their options for leaving feedback. With our own forum, this problem would mostly be solved as, depending on how we handle logging in, users could create Scratch Addons accounts (maybe) or almost everyone coming in could verify ownership of a Scratch account to authenticate and make posts, **letting more users join in**.

Allowing users to anonymously leave feedback without logging in could also be considered. Such feedback could be posted privately (only maintainers could read it) or made public as an anonymous post (though, that could potentially lead to abuse).

#### Scratch
This would connect the Scratch community to our feedback forum by verifying ownership of Scratch accounts. Everyone on the feedback forum would **inherit their username from Scratch** this way.

We shouldn't reference our website from Scratch (since our feedback forum's communication feature would very likely violate [Scratch's website policies](https://scratch.mit.edu/discuss/topic/4421/) and of course, [we don't want to mention our browser extension](https://scratch.mit.edu/discuss/topic/284272/)).

#### GitHub
This might make it easier to grant some collaborators/organization members privileges. It also provides **two-factor authentication**. GitHub has already gradually been rolling out a 2FA requirement, so we might as well take that extra protection just in case.

#### Our own account system
This might work alright if we're okay with potentially not recognizing users from Scratch. However, **users may not want to create an account just for this**.

---

@WorldLanguages I know you [previously mentioned](https://github.com/ScratchAddons/ScratchAddons/issues/5502#issue-1513118309) the idea of a feedback forum.

**Comment by WorldLanguages** [comment]
The main obstacle for this is trying to balance privacy, ease of use, practicability and avoiding abuse. For example: allowing users to upload files, such as screenshots. And don't forget that moderation will take effort.

Still, it would be beautiful to let users upvote suggestions and move non-technical discussions away from our GitHub issue boards.

**Comment by BroJac5246** [comment]
This could be very useful. It sounds like a lot happens on Discord, but some people (myself included) don't use it. For authentication, I think we should use [Scratch Auth](https://auth.itinerary.eu.org/) because it's easy and safe.

**Comment by Secret-chest** [comment]
The accounts should also have GH profile links

**Comment by Secret-chest** [comment]
I just hope it's going to be a real forum, not something like feedback.minecraft.net.

**Comment by WorldLanguages** [comment]
My idea was to take most inspiration out of https://feedback.fider.io/ and have it replace https://scratchaddons.com/feedback (feedback form and feedback forum all at once).

It's not that simple because we'd want to let users send private feedback still, but it means we can immediately let other users upvote feedback suggestions from other users. It also gets tricky when users send feedback in other languages and whether we want to localize the strings in the feedback forum. We would also need to moderate for common sense as well as avoiding duplicate suggestions.
It would also be ideal if private feedback could be sent without logging in (to replicate the current friction-less experience of the feedback form), but if we open other actions such as commenting or upvoting anonymously, that can probably go wrong.

**Comment by WorldLanguages** [comment]
Yeah speaking of moderation...
Someone just sent
<details> 
  <summary></summary>
   "Fuck you suck my long monkey cock"
</details> on the feedback form 👍

**Comment by BroJac5246** [comment]
> Yeah speaking of moderation...
> Someone just sent
> _[Aaaah I don't want to see it while quoting that]_
> on the feedback form 👍 

Was it anonymous? Why do we allow anonymous feedback in the first place? Should we continue to allow anonymous feedback?

**Comment by Secret-chest** [comment]
The example you gave is too limited, there's no replying to replies, no formatting, nothing basically.

**Comment by WorldLanguages** [comment]
> Was it anonymous? Why do we allow anonymous feedback in the first place? Should we continue to allow anonymous feedback?

Yes, it's not really problematic, it ends up in a private Discord channel as of now.

**Comment by WorldLanguages** [comment]
> The example you gave is too limited, there's no replying to replies, no formatting, nothing basically.

There's no "replying to replies" here at GitHub either. Not really a problem.
Also, Fider allows some Markdown in comments, for example you can do bold with `**asterisks**`.

I believe Fider (with some "addons" on top) can become usable for our use case.

**Comment by Secret-chest** [comment]
@WorldLanguages There's quoting here at GitHub. There should be quoting, full markdown and reactions.

**Comment by WorldLanguages** [comment]
Hmm haven't considered reactions. I doubt there's any reasonable way of adding reactions to existing feedback systems that don't support them out of the box.

**Comment by Hans5958** [comment]
I'm still thinking that something like Fider would be appropriate solely for suggestions/bugs/ideas solely by popular vote. If we want a focused experience (handling user comments/concerns/etc), then the feedback form/Discord server/GitHub Discussion would still appropriate.

**Comment by Secret-chest** [comment]
Well, can't we just get a forum (a Discourse maybe)?

**Comment by Secret-chest** [comment]
> You can post comments under each post though...?

What you can't do is comment the comments though.

**Comment by Secret-chest** [comment]
> Old-school forums and GitHub also don't have replies to replies, instead you quote the post.

You don't understand. I don't want reddit, but on normal forums you can quote.

**Comment by Secret-chest** [comment]
Fider != a forum. I do not like this ballot box system. On GitHub we can discuss the suggestion, suggest alternatives, point out problems...

The comments on the other feedback sites are a bit too small, if you get what I'm saying. They don't do anything, since they are outweighted by the OP.

**Comment by Secret-chest** [comment]
Also, Fider is incomplete. The first suggestion listef is "Allow custom suggestion templates"

**Comment by Secret-chest** [comment]
I wouldn't recommend going with a closed-source option (I know Fider is not) since then we're at the mercy of the company.

**Comment by WorldLanguages** [comment]
> The comments on the other feedback sites are a bit too small, if you get what I'm saying. They don't do anything, since they are outweighted by the OP.

Ideally, mods/admins would edit the original post to reflect discussion after the suggestion was first posted. You're probably not saying "small" literally, but we can tweak the CSS as we want.

> The first suggestion is "Allow custom suggestion templates"

Hmm, which templates do you have in mind?

> Oh, and private feedback is not supported.

It can be emulated using tags and restricting posts without the "public" from public view.

> Moreover, you only have 1 hour to edit your posts.

Didn't notice that, do you have the source? That's worth considering.

> I wouldn't recommend going with a closed-source option (I know Fider is not) since then we're at the mercy of the company.

We would probably fork Fider in some way, so we would do fine as long as they provide security patches.

**Comment by Secret-chest** [comment]
> Ideally, mods/admins would edit the original post to reflect discussion after the suggestion was first posted. You're probably not saying "small" literally, but we can tweak the CSS as we want.

The comments are crippled, that's what I mean.

> Hmm, which templates do you have in mind?

New addon, setting, enhancement, etc. It was just an example of how incomplete it is.

> Didn't notice that, do you have the source? That's worth considering.

From a recent blog:
> ### #3: Grace period to edit posts

> This was a popular feature request for a long time and we're happy to announce that users now have 1 hour to edit their posts. This should allow users to enter additional information or reformat their post after seeing it live.
